network/address_test.go

```diff
@@ -537,6 +537,73 @@ func (s *AddressSuite) TestSelectInternalMachineAddress(c *gc.C) {
 	}
 }
 
+type selectInternalAddressesTest struct {
+	about           string
+	addresses       []network.Address
+	machineLocal    bool
+	expectedIndices []int
+	expectedOk      bool
+}
+
+// expected returns the expected address for the test.
+func (t selectInternalAddressesTest) expected() ([]network.Address, bool) {
+	if t.expectedIndices == nil {
+		return nil, t.expectedOk
+	}
+	expected := make([]network.Address, len(t.expectedIndices))
+	for i, idx := range t.expectedIndices {
+		expected[i] = t.addresses[idx]
+	}
+	return expected, t.expectedOk
+}
+
+var selectInternalAddressesTests = []selectInternalAddressesTest{
+	{
+		"machine-local addresses are selected when machineLocal is true",
```

Comment:
this would probably be easier to read if you annotated the variables. Especially since you have a bare "true" in there.

about: "blah",
machineLocal: true,
etc.

as for behavior, I thought the idea was that we would select 'cloud-local' but if machine-local is true, we would fall back to machine local addresses, not that we would prefer machine local addresses if the flag is set.

maybe that's wrong in the case of a single controller with multiple addresses, that never actually goes into HA.